nbconvert/exporters/script.py

```diff
@@ -19,15 +18,29 @@ def _template_file_default(self):
 
     def from_notebook_node(self, nb, resources=None, **kw):
         langinfo = nb.metadata.get('language_info', {})
-        
         # delegate to custom exporter, if specified
         exporter_name = langinfo.get('nbconvert_exporter')
         if exporter_name and exporter_name != 'script':
             self.log.debug("Loading script exporter: %s", exporter_name)
-            from .export import get_exporter
+            from .export import get_exporter, ExporterNameError
             if exporter_name not in self._exporters:
-                Exporter = get_exporter(exporter_name)
-                self._exporters[exporter_name] = Exporter(parent=self)
+                try: 
+                    Exporter = get_exporter(exporter_name)
+                    self._exporters[exporter_name] = Exporter(parent=self)
+                except ValueError:
+                    if langinfo.get('nbconvert_exporter_package',""):
+                        external_exporter_name = langinfo.get('nbconvert_exporter_package')
+                        try: 
+                            Exporter = get_exporter(external_exporter_name)
```

Comment:
Right - `nbconvert_exporter` specifies the entry point, and `nbconvert_exporter_package` is the package which includes the entry point. The latter doesn't need to be used programmatically, it's just for showing information to the user.

We think we've come up with a better alternative below, but I think we'd benefit from taking a step back from this for a day or two before we implement either.